core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1300,24 +1298,31 @@ public NameResolverRegistry getNameResolverRegistry() {
     public void onResult(final ResolutionResult resolutionResult) {
       final class NamesResolved implements Runnable {
 
-        @SuppressWarnings("ReferenceEquality")
+        @SuppressWarnings({"ReferenceEquality", "unchecked"})
         @Override
         public void run() {
           List<EquivalentAddressGroup> servers = resolutionResult.getAddresses();
+          if (servers.isEmpty() && lastResolutionResultWasSuccess == Boolean.TRUE) {
+            channelLogger.log(
+                ChannelLogLevel.DEBUG, "Service config for no address detected, ignoring it");
+            return;
+          }
           Attributes attrs = resolutionResult.getAttributes();
           channelLogger.log(
               ChannelLogLevel.DEBUG, "Resolved address: {0}, config={1}", servers, attrs);
 
-          if (haveBackends == null || !haveBackends) {
+          if (lastResolutionResultWasSuccess == null || !lastResolutionResultWasSuccess) {
             channelLogger.log(ChannelLogLevel.INFO, "Address resolved: {0}", servers);
-            haveBackends = true;
+            lastResolutionResultWasSuccess = true;
           }
 
           nameResolverBackoffPolicy = null;
+          ConfigOrError configOrError = resolutionResult.getServiceConfig();
+          Map<String, ?> serviceConfig = null;
+          if (configOrError != null && configOrError.getConfig() != null) {
+            serviceConfig = (Map<String, ?>) configOrError.getConfig();
+          }
```

Comment:
How about:
```java
Map<String, ?> serviceConfig = null;
Status serviceConfigError = null;
if (configOrError != null) {
  serviceConfig = (Map<String, ?>) configOrError.getConfig();
  serviceConfigError = configOrError.getError();
}
```

Then the code below on line 1344 can just use `serviceConfigError` and that may be more readable.

Reply:
done.